web/src/lib/components/asset-viewer/actions/rotate-action.svelte

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+  import MenuOption from '$lib/components/shared-components/context-menu/menu-option.svelte';
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
+  import { AssetAction } from '$lib/constants';
+  import { handleError } from '$lib/utils/handle-error';
+  import { AssetJobName, ExifOrientation, runAssetJobs, updateAsset, type AssetResponseDto } from '@immich/sdk';
+  import { mdiRotateLeft, mdiRotateRight } from '@mdi/js';
+  import { t } from 'svelte-i18n';
+  import type { OnAction } from './action';
+  import { photoViewer } from '$lib/stores/assets.store';
+
+  export let asset: AssetResponseDto;
+  export let onAction: OnAction;
+  export let to: 'left' | 'right';
+
+  const handleRotateAsset = async () => {
+    // every rotation, in order
+    // interleaved normal / mirrored
+    const orientations = ['1', '2', '6', '7', '3', '4', '8', '5'];
+    let index = orientations.indexOf(asset.exifInfo?.orientation ?? '1');
+    if (index === -1) {
+      index = 0;
+    }
+    if (to === 'right') {
+      index = (index + 2) % orientations.length;
+    } else {
+      index = (index - 2 + orientations.length) % orientations.length;
+    }
+    const newOrientation = orientations[index] as ExifOrientation;
+    try {
+      await updateAsset({ id: asset.id, updateAssetDto: { orientation: newOrientation } });
+      asset.exifInfo = { ...asset.exifInfo, orientation: newOrientation };
+      await runAssetJobs({ assetJobsDto: { assetIds: [asset.id], name: AssetJobName.RegenerateThumbnail } });
+      notificationController.show({
+        type: NotificationType.Info,
+        message: $t('edited_asset'),
+      });
+
+      onAction({ type: AssetAction.ROTATE, asset });
+      setTimeout(() => {
+        // force the image to refresh the thumbnail
+        const oldSrc = new URL($photoViewer!.src);
+        oldSrc.searchParams.set('t', Date.now().toString());
+        $photoViewer!.src = oldSrc.toString();
```

Comment:
We rely on the checksum field to detect changes in the thumbnail and bypass the cache. Refreshing based on the time might work in this component, but it will cause caching issues everywhere else.

Reply:
I agree that it's not pretty; but @jrasm91 told me on Discord the ideal way would through a web socket event but that since we don't currently have one for thumbnail generation, refreshing like that was a good enough hack until it's done. 

How would you do it?

Reply:
I don't see how a websocket event will solve caching issues in other components/pages. We include the `checksum` for thumbnail requests, but the checksum doesn't change in this case so the cached thumbnail is still used.

Reply:
Maybe we should use a counter/nonce instead of the checksum

Reply:
Oh sorry now I get your point. You're absolutely right, yes.

Reply:
so, uh, sorry, but what should I change here?